### ISSUE TITLE
Refactor abstraction.py

### ISSUE THREAD

**Issue by sdemarch** [NeVerTools/pyNeVer#25]
Restructure the module abstraction.py to handle abstract interpretation objects and methods

- (perhaps) create new module for Star and StarSet classes
- Remove AbsNetwork classes, either move the forward and backwards entry points in the concrete network class or create new class for handling at a high level the abstracion flow
- Refactor functions into (weak-private) class methods for AbsLayerNodes now that Python allows to use multiprocessing on them

**Comment by darioguidotti** [comment]
It might make sense to transform `AbsNeuralNetwork` into a wholly abstract class, requiring the implementation of `forward` and `backward` methods. Subsequently, we would have `AbsSeqNetwork` and `AbsAcyclicNetwork`, both inheriting from `AbsNeuralNetwork` as well as their respective counterparts in `networks.py` (`SequentialNetwork` and `AcyclicNetwork`). This approach ensures the inheritance of all node management methods, mitigating redundancy in the codebase.

To make this approach work we will need to complete the planned refactoring in node.py (#24) and networks.py (#23).

**Comment by sdemarch** [comment]
Should it be the case that, instead of a method `build_abs_network` in `verification.py`, the same procedure is the abstract class constructor in `abstraction.py`? It is (and should) never the case that an abstract network is initialized without the complete concrete network.

TL;DR `AbsSeqNetwork` init method should contain the code of `__build_abs_seq_network`?

**Comment by sdemarch** [comment]
We decided to structure the module as follows:

- Make `abstraction` a package with submodules `networks` for abstract elements and networks, and `layers` for abstract layers
- Make a base class `LayerNode` with only the identifier as a parameter, and have `AbsLayerNode` and `ConcreteLayerNode` objects inherit from it (`nodes.py`). `AbsLayerNode` will contain a reference to their concrete counterpart, and the `forward` and `backward` methods (`abstraction.layers.py`).
- Make a base class `NeuralNetwork` representing a graph, and the classes `SequentialNetwork` and `AcyclicNetwork` as children specializing it (`networks.py`). 
- Make an interface `IAbstractNetwork` to provide `forward` and `backward` methods to concrete classes `AbsSequentialNetwork` and `AbsAcyclicNetwork` which will inherit from the interface and their concrete counterpart.
- Make a Factory for abstract networks instead of directly instantiating them
